blosc/shuffle-altivec.c

```diff
@@ -15,61 +16,145 @@
 #endif
 
 #include <emmintrin.h>
-
-
-/* The next is useful for debugging purposes */
-#if 0
+#include <altivec.h>
 #include <stdio.h>
-#include <string.h>
 
-static void printxmm(__m128i xmm0)
+
+//Store a vector to an unaligned location in memory
+static void vec_st_unaligned(__vector uint8_t v, int32_t position,  __vector uint8_t * dst)
 {
-  uint8_t buf[16];
-
-  ((__m128i *)buf)[0] = xmm0;
-  printf("%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x\n",
-          buf[0], buf[1], buf[2], buf[3],
-          buf[4], buf[5], buf[6], buf[7],
-          buf[8], buf[9], buf[10], buf[11],
-          buf[12], buf[13], buf[14], buf[15]);
+    //Load the surrounding area
+    __vector uint8_t low = vec_ld( position, dst);
+    __vector uint8_t high = vec_ld( position+16, dst);
+    //Prepare the constants that we need
+    uint8_t offset = ((size_t)dst) % 16;
+    __vector uint8_t permute_lo, permute_hi;
+    for (uint8_t i=0; i<16; i++)
+    {
```

Comment:
I typically prefer the curly brace to be at the end of `for` statement (kind of similar to the colon at the end of a `for` in Python).